common-lib/src/main/scala/com/gu/mediaservice/model/ThrallMessage.scala

```diff
@@ -1,88 +1,21 @@
-  implicit val updateImageSyndicationMetadataMessageFormat = Json.format[UpdateImageSyndicationMetadataMessage]
-  implicit val setImageCollectionsMessageFormat = Json.format[SetImageCollectionsMessage]
-  implicit val updateImageUserMetadataMessageFormat = Json.format[UpdateImageUserMetadataMessage]
-  implicit val deleteImageExportsMessageFormat = Json.format[DeleteImageExportsMessage]
-  implicit val imageMessageFormat = Json.format[ImageMessage]
-  implicit val updateImagePhotoshootMetadataMessage = Json.format[UpdateImagePhotoshootMetadataMessage]
-  implicit val deleteUsagesMessage = Json.format[DeleteUsagesMessage]
-  implicit val updateImageUsagesMessage = Json.format[UpdateImageUsagesMessage]
-  implicit val addImageLeaseMessage = Json.format[AddImageLeaseMessage]
-  implicit val removeImageLeaseMessage = Json.format[RemoveImageLeaseMessage]
-  implicit val updateImageExportsMessage = Json.format[UpdateImageExportsMessage]
-  implicit val writes = Json.writes[ThrallMessage]
-
-}
-
-
-
-case class ImageMessage(id: String, lastModified: DateTime, image: Image) extends ThrallMessage {
-  override def additionalMarkers: () => Map[String, Any] = ()=>
-    Map("fileName" -> image.source.file.toString)
 }
 
-case class DeleteImageMessage(id: String, lastModified: DateTime) extends ThrallMessage
-
-case class DeleteImageExportsMessage(id: String, lastModified: DateTime) extends ThrallMessage
-
-case class UpdateImageExportsMessage(id: String, lastModified: DateTime, crops: Seq[Crop]) extends ThrallMessage
-
-case class UpdateImageUserMetadataMessage(id: String, lastModified: DateTime, edits: Edits) extends ThrallMessage
-
-case class UpdateImageUsagesMessage(id: String, lastModified: DateTime, usageNotice: UsageNotice) extends ThrallMessage
-
-case class ReplaceImageLeasesMessage(id: String, lastModified: DateTime, leases: Seq[MediaLease]) extends ThrallMessage
-
-case class AddImageLeaseMessage(id: String, lastModified: DateTime, lease: MediaLease) extends ThrallMessage
-
-case class RemoveImageLeaseMessage(id: String, lastModified: DateTime, leaseId: String) extends ThrallMessage
-
-case class SetImageCollectionsMessage(id: String, lastModified: DateTime, collections: Seq[Collection]) extends ThrallMessage
-
-case class DeleteUsagesMessage(id: String, lastModified: DateTime) extends ThrallMessage
-
-object DeleteUsagesMessage {
-  implicit val yourJodaDateReads = JodaReads.DefaultJodaDateTimeReads.map(d => d.withZone(DateTimeZone.UTC))
-  implicit val yourJodaDateWrites = JodaWrites.JodaDateTimeWrites
-  implicit val what = Json.format[DeleteUsagesMessage]
-}
-
-case class UpdateImageSyndicationMetadataMessage(id: String, lastModified: DateTime, maybeSyndicationRights: Option[SyndicationRights]) extends ThrallMessage
-
-case class UpdateImagePhotoshootMetadataMessage(id: String, lastModified: DateTime, edits: Edits) extends ThrallMessage
-
-
-
+case class CreateMigrationIndexMessage(migrationStart: DateTime ,gitHash: String)
```

Comment:
as per PR description the new index will be called `images_YYYY-MM-DDTHH:mm:ss_GITHASH` (this case class looks clearly related).
I like that we're putting a traceable unique reference in the new index name, but since these migrations could last many days (even weeks) does this mean that we can't release other grid things while a migration is underway, since this would then create a further new index and essentially start again - apologies if I've misunderstood what this git hash refers to

Reply:
the hash/date is just for convenience, so we can see when it was started.

Reply:
ahh I see, because it's checking for the consistently named `Images_Migration` alias - gotcha